backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/KeycloakApiService.java

```diff
@@ -219,4 +219,14 @@ private ResponseEntity<Object> deleteUserIdentityProviderLink(String userId, Str
     private boolean identityProviderLinkDeletedSuccessfully(List<ResponseEntity<Object>> deleteIdentityProviderLinkResponses) {
         return deleteIdentityProviderLinkResponses.stream().allMatch(deleteIdentityProviderLinkResponse -> deleteIdentityProviderLinkResponse.getStatusCode().equals(HttpStatus.NO_CONTENT));
     }
+
+    public List<Object> bulkRemoveUserClientRoles(String clientGuid, BulkRemovalRequest bulkRemovalRequest) {
+        List<Object> responseList = new ArrayList<>();
+        if (getClient(clientGuid).getStatusCode().is2xxSuccessful()) {
+            bulkRemovalRequest.getUserRolesForRemoval().forEach((userId, rolesToDelete) -> responseList.add(deleteUserClientRole(userId, clientGuid, rolesToDelete)));
```

Comment:
Not in this PR, but if we're removing 100+ users, it will probably be nice to use https://spring.io/guides/gs/async-method

Reply:
Thanks, I will take a look at that